scripts/lib.sh

```diff
@@ -251,22 +273,86 @@ generate_dockerfile()
 
 	curlOptions=("-OL")
 	[ -n "${http_proxy:-}" ] && curlOptions+=("-x ${http_proxy:-}")
+
+	readonly dockerfile_template="Dockerfile.in"
 	readonly install_go="
 RUN cd /tmp ; curl ${curlOptions[@]} https://storage.googleapis.com/golang/go${GO_VERSION}.linux-${goarch}.tar.gz
 RUN tar -C /usr/ -xzf /tmp/go${GO_VERSION}.linux-${goarch}.tar.gz
 ENV GOROOT=/usr/go
 ENV PATH=\$PATH:\$GOROOT/bin:\$GOPATH/bin
 "
 
-	readonly dockerfile_template="Dockerfile.in"
+	# Rust agent
+	# rust installer should set path apropiately, just in case
+	local cmake_file="cmake-${CMAKE_VERSION}.tar.gz"
+	local cmake_dir="cmake-${CMAKE_VERSION}"
+	readonly install_cmake="
+RUN pushd /root; \
+    curl -sLO https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/${cmake_file}; \
+	tar -zxf ${cmake_file}; \
+	cd ${cmake_dir}; \
+	./bootstrap; \
+	make; \
+	make install; \
+	popd
+"
+	local musl_tar="musl-${MUSL_VERSION}.tar.gz"
+	local musl_dir="musl-${MUSL_VERSION}"
+	readonly install_musl="
+RUN pushd /root; \
+    curl -sLO https://www.musl-libc.org/releases/${musl_tar}; tar -zxf ${musl_tar}; \
+	cd ${musl_dir}; \
+	sed -i \"s/^ARCH = .*/ARCH = ${muslarch}/g\" dist/config.mak; \
+	./configure; \
+	make; \
+	make install; \
+	echo \"/usr/local/musl/lib\" > /etc/ld-musl-${muslarch}.path; \
+	popd
+ENV PATH=\$PATH:/usr/local/musl/bin
+"
+	readonly install_rust="
+RUN curl https://sh.rustup.rs -sSf --output /tmp/rust-init; \
```

Comment:
OOI, why are we not using the curl command specified on `rustup.rs` which specifies `--proto '=https' --tlsv1.2`. I also wonder if you should add a `-L` "just in case"?